blockchain/blockdao/blob_store.go

```diff
@@ -167,17 +147,48 @@ func (bs *blobStore) PutBlock(blk *block.Block) error {
 			pb.Sidecars = append(pb.Sidecars, action.ToProtoSideCar(b))
 		}
 	}
-	raw, err := proto.Marshal(&pb)
-	if err != nil {
-		return errors.Wrapf(err, "failed to put block = %d", blk.Height())
+	var (
```

Comment:
`putSidecars()' now calls `deleteBlob()` inside, which is not ideal, I think it's better to organize like this:
```
func (bs *blobStore) PutBlock(blk *block.Block) error {
	if blk.Height() <= atomic.LoadUint64(&bs.currWriteBlock) {
	        return errors.Errorf("block height %d is less than current tip height", blk.Height())
	}
	b := batch.NewBatch()
	if blk.HasBlob() {
	        // prepare the pb, raw (L136-L149)
	        bs.putBlob(raw, height, pb.TxHash, b)
	}
	// delete expired blob
	bs.deleteBlob(height, b)
	if err := bs.kvStore.WriteBatch(b); err != nil {
		return errors.Wrapf(err, "failed to put blob")
	}
	atomic.StoreUint64(&bs.currWriteBlock, height)
	return nil
```